jobs/jobs_api_go_sdk.go

```diff
@@ -0,0 +1,307 @@
+
+	err = StopActiveRun(jobID, a.d.Timeout(schema.TimeoutUpdate), w, ctx)
+
+	if err != nil {
+		return err
+	}
+	return Start(jobID, a.d.Timeout(schema.TimeoutUpdate), w, ctx)
+}
+
+type controlRunStateLifecycleManagerGoSdk struct {
+	d *schema.ResourceData
+	m *common.DatabricksClient
+}
+
+func (c controlRunStateLifecycleManagerGoSdk) OnCreate(ctx context.Context) error {
+	return nil
+}
+
+func (c controlRunStateLifecycleManagerGoSdk) OnUpdate(ctx context.Context) error {
+	if c.d.Get("continuous") == nil {
+		return nil
+	}
+
+	jobID, err := parseJobId(c.d.Id())
+	if err != nil {
+		return err
+	}
+
+	w, err := c.m.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+
+	// Only use RunNow to stop the active run if the job is unpaused.
+	pauseStatus := c.d.Get("continuous.0.pause_status").(string)
+	if pauseStatus == "UNPAUSED" {
+		// Previously, RunNow() was not supported for continuous jobs. Now, calling RunNow()
+		// on a continuous job works, cancelling the active run if there is one, and resetting
+		// the exponential backoff timer. So, we try to call RunNow() first, and if it fails,
+		// we call StopActiveRun() instead.
+		_, err := w.Jobs.RunNow(ctx, jobs.RunNow{
+			JobId: jobID,
+		})
+
+		if err == nil {
+			return nil
+		}
+
+		// RunNow() returns 404 when the feature is disabled.
+		var apiErr *apierr.APIError
+		if errors.As(err, &apiErr) && apiErr.StatusCode != 404 {
+			return err
+		}
+	}
+
+	return StopActiveRun(jobID, c.d.Timeout(schema.TimeoutUpdate), w, ctx)
+}
+
+// Removing unnecesary fields out of ClusterSpec's ForceSendFields because the current terraform plugin sdk
+// has a limitation that it will always set unspecified values to the zero value.
+func removeUnnecessaryFieldsFromForceSendFields(clusterSpec *compute.ClusterSpec) error {
+	newForceSendFields := []string{}
+	for _, field := range clusterSpec.ForceSendFields {
+		// `autotermination_minutes` should never be explicitly set for jobs because automated clusters don't accept this config.
+		// TODO: consider removing `autotermination_minutes` from the jobs terraform schema.
+		if field != "AutoterminationMinutes" {
+			newForceSendFields = append(newForceSendFields, field)
+		}
+	}
+	clusterSpec.ForceSendFields = newForceSendFields
+
+	if clusterSpec.AwsAttributes != nil {
+		newAwsAttributesForceSendFields := []string{}
+		// These fields should never be 0.
+		unnecessaryFieldNamesForAwsAttributes := []string{
+			"SpotBidPricePercent",
+			"EbsVolumeCount",
+			"EbsVolumeIops",
+			"EbsVolumeSize",
+			"EbsVolumeThroughput",
+		}
+		for _, field := range clusterSpec.AwsAttributes.ForceSendFields {
+			if !slices.Contains(unnecessaryFieldNamesForAwsAttributes, field) {
+				newAwsAttributesForceSendFields = append(newAwsAttributesForceSendFields, field)
+			}
+		}
+		clusterSpec.AwsAttributes.ForceSendFields = newAwsAttributesForceSendFields
+	}
+	return nil
+}
+
+func prepareJobSettingsForUpdateGoSdk(d *schema.ResourceData, js JobSettingsResource) {
```

Comment:
This should be `*JobSettingsResource`?

Reply:
Looks like we don't have to, but yeah changing it to be safe